### DISCUSSION TITLE
Speed accuracy (was I skiing at 185km/h speed?(!))

### DISCUSSION THREAD

**Discussion by chipechop** [OpenTracksApp/OpenTracks#1135] [Q&A]
Hi guys, 
How would you set the GPS settings, to correctly measure the speed of a skier, while skiing?

I don't know if it was a problem of the GPS settings, but the app measured an unbelievable 185km/h maximum speed which is simply above human capabilities...

What would you check in the settings of the app, to be sure it measured speed correctly? 

Is it, actually, a problem in the settings that could lead to such an abnormal measure?

**Comment by dennisguse** [comment]
Sadly not (yet).

The problem is kind of manifold.
First of all, OpenTracks uses the speed estimate provided by the GPS.
And then we apply some filtering as GPS is known be kind of imprecise regarding speed.
Especially, if the GPS reception is not great (even with great reception, it is not great) - like in mountains.
I am even not sure if GPS takes vertical speed (i.e., losing altitude while skiing) into account.
Anyhow, vertical accuracy is not great with GPS...

The plan was to change OpenTracks to compute the speed based upon the traveled distance.
But sadly that is not that straight forward and nobody had time to do it.
@vlmendz wrote down some thoughts here: https://github.com/OpenTracksApp/OpenTracks/issues/1003

@chipechop One question: while skiing would you compute the speed in 3D (i.e., including altitude loss)?
